### PR TITLE
docs: refactor doc files to present better doc pages

### PR DESCRIPTION
**Pull request by ChongWei905**
Thank you for your contribution to the MindYOLO repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindyolo/blob/master/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by Ash-Lee233 — DISMISSED** [review]

File: docs/zh/tutorials/configuration.md [hunk 1]
```diff
@@ -2,4 +2,249 @@
...
+     [-1, 1, Bottleneck, [64]],
+     [-1, 1, ConvNormAct, [128, 3, 2]],  # 3-P2/4
+     [-1, 2, Bottleneck, [128]],
+     [-1, 1, ConvNormAct, [256, 3, 2]],  # 5-P3/8
+     [-1, 8, Bottleneck, [256]],
+     [-1, 1, ConvNormAct, [512, 3, 2]],  # 7-P4/16
+     [-1, 8, Bottleneck, [512]],
+     [-1, 1, ConvNormAct, [1024, 3, 2]],  # 9-P5/32
+     [-1, 4, Bottleneck, [1024]],  # 10
+    ]
```
Comment (zhanghuiyao) [inline docs/zh/tutorials/configuration.md:157]: 建议后面可以把这块详细解读下，构建的方式，每个块的连接方式，可以增加一些例子图片帮助理解，另外不通过backbone/head的方式直接定义所需模型的方式后面也可以加一下

File: docs/en/how_to_guides/callback.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+The flexibility and extensibility of the program can be increased by using the callback function, because users can register custom methods to the mount point to be called without modifying the code in the program.
+
+In MindYOLO, the callback function is specifically implemented in the mindyolo/utils/callback.py file.
+```python
+#mindyolo/utils/callback.py
+@CALLBACK_REGISTRY.registry_module()
+class callback_class_name(BaseCallback):
+
+def __init__(self, **kwargs):
+super().__init__()
```
Comment (CaitinZhao) [inline docs/en/how_to_guides/callback.md:13]: python的格式注意下

File: docs/en/how_to_guides/callback.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Usage of MindYOLO callback function
```
Comment (CaitinZhao) [inline docs/en/how_to_guides/callback.md:1]: 可以给个库上已经实现的代码的链接，然后下面给怎么自定义实现

File: docs/en/how_to_guides/data_preparation.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Data preparation
+
+## Dataset format introduction
+
+Download coco2017 YOLO format [coco2017labels-segments](https://github.com/ultralytics/yolov5/releases/download/v1.0/coco2017labels-segments.zip) and coco2017 original images [train2017](http://images.cocodataset.org/zips/train2017.zip) , [val2017](http://images.cocodataset.org/zips/val2017.zip) , then put the coco2017 original images into the coco2017 YOLO format images directory:
+```
+└─ coco2017_yolo
+├─ annotations
```
Comment (CaitinZhao) [inline docs/en/how_to_guides/data_preparation.md:8]: 换行都注意下

File: docs/en/how_to_guides/write_a_new_model.md [hunk 5]
```diff
@@ -1,2 +1,208 @@
...
+def _cfg(url="", **kwargs):
+return {"url": url, **kwargs}
+
+default_cfgs = {"mymodel": _cfg(url="")}
+```
+#### Create a model
+In `MindSpore`, the model class inherits from `nn.Cell`. Generally, the following two functions need to be overloaded:
+
+- In the `__init__` function, the module layer needed in the model should be defined.
+- In the `construct` function, define the model forward logic. <br> ```python class MYmodel(nn.Cell): def __init__(self, cfg, in_channels=3, num_classes=None, sync_bn=False): super(MYmodel, self).__init__() self.cfg = cfg self.stride = Tensor(np.array(cfg.stride), ms.int32) self.stride_max = int(max(self.cfg .stride)) ch, nc = in_channels, num_classes self.nc = nc # override yaml value self.model = build_model_from_cfg(model_cfg=cfg, in_channels=ch, num_classes=nc, sync_bn=sync_bn) self.names = [str(i) for i in range(nc)] # default names initialize_defult() # Optional, you may need the initialize_defult method to get the same conv2d and dense layer initialization method as pytorch;
```
Comment (CaitinZhao) [inline docs/en/how_to_guides/write_a_new_model.md:139]: 这段的格式是不是乱掉了

File: docs/en/tutorials/configuration.md [hunk 6]
```diff
@@ -2,4 +2,248 @@
...
+   - per_batch_size: batch size of each card, default is 32
+   - epochs: number of training epochs, default is 300
+   - ...
+
+
+### Parse parameter settings
+This part of the parameters is usually passed in from the command line. Examples are as follows:
+
+  ```shell
+  mpirun --allow-run-as-root -n 8 python train.py --config ./configs/yolov7/yolov7.yaml  --is_parallel True --log_interval 50
```
Comment (CaitinZhao) [inline docs/en/tutorials/configuration.md:45]: 2.3的话最好多卡启动都改成msrun

File: docs/en/tutorials/data_augmentation.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    - {func_name: image_norm, scale: 255.}
+    - {func_name: image_transpose, bgr2rgb: True, hwc2chw: True}
+```
+_Note: func_name represents the name of the data enhancement method, prob, mosaic9_prob, translate, and scale are the method parameters. Among them, prob is a parameter common to all methods, indicating the execution probability of the data enhancement method. The default value is 1_
+
+The specific operations performed by the above yaml file are as follows:
+
+- `mosaic`: Perform mosaic operation on the input image with a probability of 1.0, that is, splicing 4 different images into one image. mosaic9_prob represents the probability of splicing using the 9-square grid method, and translate and scale represent the degree of random translation and scaling respectively.
+as the picture shows:
+<img width='600' src="https://github.com/ChongWei905/images/raw/main/mindyolo-mosaic-en.png"/>
```
Comment (CaitinZhao) [inline docs/en/tutorials/data_augmentation.md:59]: 这个页面多放一点图

File: docs/en/tutorials/deployment.md [hunk 8]
```diff
@@ -1,5 +1,254 @@
+# Deployment
 
+## Dependencies
+   ```shell
+   pip install -r requirement.txt
+   ```
 
-# Deployment
+## MindSpore Lite environment preparation
```
Comment (CaitinZhao) [inline docs/en/tutorials/deployment.md:8]: lite这个由于最近几个版本都没有验过了，把之前验过的mindspore和lite的版本写清楚

**Review by zhanghuiyao — DISMISSED** [review]

File: docs/en/how_to_guides/write_a_new_model.md (hunk 5 above)
Comment (ChongWei905) [reply docs/en/how_to_guides/write_a_new_model.md]: 已修复

File: docs/en/how_to_guides/data_preparation.md (hunk 4 above)
Comment (ChongWei905) [reply docs/en/how_to_guides/data_preparation.md]: 已修复

File: docs/en/how_to_guides/callback.md (hunk 3 above)
Comment (ChongWei905) [reply docs/en/how_to_guides/callback.md:1]: 已新增

File: docs/en/how_to_guides/callback.md (hunk 2 above)
Comment (ChongWei905) [reply docs/en/how_to_guides/callback.md]: 已修复

**Review by zhanghuiyao — APPROVED** [review]

**Pull request merged**